### ISSUE TITLE
多任务之中检测任务，指标不增长，始终为0

### ISSUE THREAD

**Issue by Lacacy** [JiayuanWang-JW/YOLOv8-multi-task#5]
参照readme进行的配置；请大佬帮忙看下这个问题吧

```yaml
# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
#path: /data/jiayuan/BDDcoco/yolo_v8_toy  # dataset root dir
path: /data/wkx/develop/data/new_lawn  # dataset root dir

# Train/val/test image paths for all tasks
train: images/train  # train images for object detection (relative to 'path')


val: images/val  # val images for object detection (relative to 'path')


test: images/val  # test images for object detection (relative to 'path')

labels_list:
  - detection_object
  - seg_grass
  - seg_bush

tnc: 5  # number of classes
nc_list: [3,1,1]
map: [None,{'3':'0'},{'4':'0'}]

# Classes for all tasks
names:
  0: leaf
  1: branch
  2: bottle
  3: grass  # Add drivable class for drivable segmentation
  4: brush  # Add lane class for lane segmentation
```

**Comment by Lacacy** [comment]
Class     Images  Instances      Box(P          R      mAP50   mAP50-95
                   all         61        125          0          0          0          0
                 grass     pixacc      0.894     subacc      0.864        IoU      0.746       mIoU      0.768
                 brush     pixacc      0.832     subacc      0.671        IoU      0.244       mIoU      0.524

**Comment by JiayuanWang-JW** [comment]
Please show me your train.py code. Thanks.

**Comment by Lacacy** [comment]
ok
```python
import sys
sys.path.insert(0, r"/data/wkx/develop/YOLOv8-multi-task-main")
# 现在就可以导入Yolo类了
from ultralytics import YOLO

# Load a model
model = YOLO(r'/data/wkx/develop/YOLOv8-multi-task-main/ultralytics/models/v8/yolov8-bdd-v4-one-dropout-individual-n.yaml', task='multi')  # build a new model from YAML

# Train the model
model.train(data=r'/data/wkx/develop/YOLOv8-multi-task-main/ultralytics/datasets/bdd-multi.yaml', resume=False, batch=16, epochs=300, imgsz=(640,640), device=[0], name='yolopm', val=True, task='multi', single_cls=False)

```

**Comment by JiayuanWang-JW** [comment]
Your setting looks good so far. Did you change the structure of the model?

**Comment by Lacacy** [comment]
I found out why, the detection box of this project may have a wrong correspondence with the image.
In the following figures, the red boxes are showing labeling errors. However, I confirmed that my data labels were all correct by using the YOLOv6 data visualization program myself.
![image](https://github.com/JiayuanWang-JW/YOLOv8-multi-task/assets/63200713/21b9cb3d-683f-448e-bb96-f4542e8a47ac)

**Comment by JiayuanWang-JW** [comment]
I checked the labels on the image in our dataset. That is correct. So please recheck your dataset label. Please follow this instruction to prepare your detection dataset. [Detection task dataset format](https://docs.ultralytics.com/datasets/detect/)

<img width="881" alt="image" src="https://github.com/JiayuanWang-JW/YOLOv8-multi-task/assets/105391468/a1dd6534-5daf-4eab-9c27-0a4eebf26b66">

**Comment by Lacacy** [comment]
When I use the official yolov8 program and just train the detection, the training set presentation given is also OK. And, the MAP metrics are also normal and the inference results are also correct.
Have you tried training multiple detection categories with yolov8-multi-task? I'm guessing that it might be because of my multi-category that this is happening.
![image](https://github.com/JiayuanWang-JW/YOLOv8-multi-task/assets/63200713/3b830f4a-ed9e-42c7-bdfc-c71201216e43)

**Comment by JiayuanWang-JW** [comment]
Yes. This image includes multi-task detetion. You can see the different colour boxes in the image. They are different classifications. We have four classes in the whole dataset. I am not sure if this image shows all of the classes. But I am sure there are two classes at least—Orange and yellow boxes.

> I checked the labels on the image in our dataset. That is correct. So please recheck your dataset label. Please follow this instruction to prepare your detection dataset. [Detection task dataset format](https://docs.ultralytics.com/datasets/detect/)
> 
> <img alt="image" width="881" src="https://user-images.githubusercontent.com/105391468/275110013-a1dd6534-5daf-4eab-9c27-0a4eebf26b66.png">

**Comment by Lacacy** [comment]
Can you re-upload a copy of the code?

**Comment by Lacacy** [comment]
That bug was eliminated when I disabled mosaic augment and just used rectangle training;

**Comment by JiayuanWang-JW** [comment]
I tried clone this repository and built a new environment to run the train.py with my dataset. It is going on well. I circle the different classifications in the below images. So I am sure the code is fine with my dataset. For your personal dataset. There are two possible issues. One is your labels have some issues. Another is my code can not support the image which is not 720x1280 resolution. To be honest, I didn't try the other images to train or predict. So please try your best to figure it out and share your experience. Thanks. 

![image](https://github.com/JiayuanWang-JW/YOLOv8-multi-task/assets/105391468/f5f7a679-6f05-485d-910b-ced2d502252d)
